pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -372,6 +448,18 @@ def newpkgcb(pkgId, name, arch):
                                 deferred_download=self.deferred_download
                             )
                             dc = DeclarativeContent(content=package, d_artifacts=[da])
+
+                            # find if a package relates to a modulemd
```

Comment:
You know here if package is a modular or not, so I think this is the place to add it and not later, because here the object is not saved yet (if it doesn't exist)